crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2812,14 +2812,168 @@ impl Analyzer<'_, '_> {
+                    if let Some(value) = &value.raw {
+                        if !value.to_lowercase().eq(&value.to_string()) {
+                            return Err(ErrorKind::AssignFailed {
+                                span: r.span(),
+                                left: box Type::StringMapping(to.clone()),
+                                right_ident: None,
+                                right: box r.clone(),
+                                cause: vec![],
+                            }
+                            .into());
+                        }
+                    }
+                }
+                IntrinsicKind::Capitalize => {
+                    if let Some(value) = &value.raw {
+                        let ch = value.to_string().chars().next();
+
+                        if let Some(ch) = ch {
+                            if !ch.is_uppercase() {
+                                return Err(ErrorKind::AssignFailed {
+                                    span: r.span(),
+                                    left: box Type::StringMapping(to.clone()),
+                                    right_ident: None,
+                                    right: box r.clone(),
+                                    cause: vec![],
+                                }
+                                .into());
+                            }
+                        }
+                    }
+                }
+                IntrinsicKind::Uncapitalize => {
+                    if let Some(value) = &value.raw {
+                        let ch = value.to_string().chars().next();
+
+                        if let Some(ch) = ch {
+                            if !ch.is_lowercase() {
+                                return Err(ErrorKind::AssignFailed {
+                                    span: r.span(),
+                                    left: box Type::StringMapping(to.clone()),
+                                    right_ident: None,
+                                    right: box r.clone(),
+                                    cause: vec![],
+                                }
+                                .into());
+                            }
+                        }
+                    }
+                }
+            },
+            Type::StringMapping(string) => {
+                if string.kind != to.kind {
+                    return Err(ErrorKind::AssignFailed {
+                        span: r.span(),
+                        left: box Type::StringMapping(to.clone()),
+                        right_ident: None,
+                        right: box r.clone(),
+                        cause: vec![],
+                    }
+                    .into());
+                }
+
+                return Ok(());
+            }
+            Type::Tpl(tpl) => match to.kind {
+                IntrinsicKind::Uppercase => {
+                    let is_not_uppercase = tpl.quasis.iter().all(|s| s.value == s.value.to_uppercase());
```

Comment:
You can do `s.value.is_uppercase()`

Reply:
And I think the variable name is confusing

Reply:
I dont think that I can do `s.value.is_uppercase` because `s.value` is an atom but let me know if Im missing something. In any case I want to re-verify the logic for template literals so please let me do that and I'll ask for a review again :pray:

Reply:
NVM, Im positive that `Uppercase<tpltype>` and `Lowercase<tpltype>` are affected by https://github.com/dudykr/stc/issues/771 so I ll just correct the if conditions for now

Reply:
See: https://doc.rust-lang.org/book/ch15-02-deref.html

The methods of `&str` can be used for atom types, too.

Reply:
Oh I asked for review before I saw this comment, I ll do that when I get back, sorry for the noise!

Reply:
@kdy1 I apologize if I'm wrong but I don't think `&str` has `.is_upperacase()` only  `to_uppercase()`.... right? :sweat_smile:

Reply:
If so you can use .chars().all(|c| ...)

Reply:
done :P